rbd/admin/task_test.go

```diff
@@ -0,0 +1,423 @@
+					PoolName:      "pool",
+					PoolNamespace: "",
+					ImageName:     "",
+					ImageID:       "12345",
+				},
+				InProgress:    false,
+				Progress:      0,
+				RetryAttempts: 1,
+				RetryTime:     "2021-09-13T05:58:24.826408",
+				RetryMessage:  "",
+			},
+			wantErr: false,
+		},
+		{
+			name: "",
+			args: args{
+				res: commands.NewResponse([]byte(tr2), "", nil),
+			},
+			want: TaskResponse{
+				Sequence: 2,
+				ID:       "id-2",
+				Message:  "Removing image pool/test",
+				Refs: TaskRefs{
+					Action:        "remove",
+					PoolName:      "pool",
+					PoolNamespace: "",
+					ImageName:     "test",
+					ImageID:       "123456",
+				},
+				InProgress:    true,
+				Progress:      0.70,
+				RetryAttempts: 0,
+				RetryTime:     "",
+				RetryMessage:  "",
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := parseTaskResponse(tt.args.res)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("parseTaskResponse() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("parseTaskResponse() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TesParseTaskResponseList(t *testing.T) {
+	type args struct {
+		res commands.Response
+	}
+	res1, err := parseTaskResponse(commands.NewResponse([]byte(tr1), "", nil))
+	assert.NoError(t, err)
+	res2, err := parseTaskResponse(commands.NewResponse([]byte(tr2), "", nil))
+	assert.NoError(t, err)
+
+	tests := []struct {
+		name    string
+		args    args
+		want    []TaskResponse
+		wantErr bool
+	}{
+		{
+			name: "emptyList",
+			args: args{
+				res: commands.NewResponse([]byte(`[]`), "", nil),
+			},
+			want:    []TaskResponse{},
+			wantErr: false,
+		},
+		{
+			name: "twoItemList",
+			args: args{
+				res: commands.NewResponse([]byte(trList), "", nil),
+			},
+			want:    []TaskResponse{res1, res2},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := parseTaskResponseList(tt.args.res)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("parseTaskResponseList() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("parseTaskResponseList() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+var (
+	imageName = "img"
+	testNS    = "test"
+)
+
+type args struct {
+	pool      string
+	namespace string
+	imageName string
+}
+
+var tests = []struct {
+	name string
+	args args
+}{
+	{
+		name: "onlyImageName",
+		args: args{
+			pool:      "",
+			namespace: "",
+			imageName: imageName,
+		},
+	},
+	{
+		name: "Image&PoolName",
+		args: args{
+			pool:      defaultPoolName,
+			namespace: "",
+			imageName: imageName,
+		},
+	},
+	{
+		name: "AllArgs",
+		args: args{
+			pool:      defaultPoolName,
+			namespace: testNS,
+			imageName: imageName,
+		},
+	},
+}
+
+func TestTaskAdmin_AddRemove(t *testing.T) {
```

Comment:
Underscore? I'm guessing that revive isn't complaining because this is test code, but it looks out of place. Let's be consistent with the rest of go-ceph and drop the underscores here and in the other function names.